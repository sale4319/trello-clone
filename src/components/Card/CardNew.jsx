import React, { useEffect, useState } from 'react';
import { useCreateCard } from '../../api/apiHooks/apiCards';

const CardNew = ({ show, listId, handleClose }) => {
    const [newCard, setNewCard] = useState('');
    const [submitNewCard, setSubmitNewCard] = useState('');
    const { isSuccess } = useCreateCard(listId, submitNewCard);

    useEffect(() => {
        if (isSuccess && submitNewCard) {
            setSubmitNewCard('');
            setNewCard('');
        }
    }, [isSuccess, submitNewCard]);

    const handleNewCardBlur = () => {
        if (newCard) {
            setSubmitNewCard(newCard);
        }
        handleClose();
    };

    const handleNewCardKeyPressEnter = (event) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            setSubmitNewCard(newCard);
            handleClose();
        }
    };

    const handleNewCardKeyPressEscape = (event) => {
        if (event.key === 'Escape') {
            handleClose();
        }
    };
    return (
        <>
            {show && (
                <textarea
                    value={newCard}
                    className="card-text-area"
                    onChange={event => setNewCard(event.target.value)}
                    autoFocus
                    onBlur={handleNewCardBlur}
                    onKeyPress={handleNewCardKeyPressEnter}
                    onKeyDown={handleNewCardKeyPressEscape}
                />
            )}
        </>
    );
};

export default CardNew