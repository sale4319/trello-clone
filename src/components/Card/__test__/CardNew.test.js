import React from 'react';
import CardNew from '../CardNew';

import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

test('component renders', () => {
    renderer.create(<CardNew />).getInstance();

  });

test("mateches snapshot", () => {
    const tree = renderer.create(<CardNew/>).toJSON();
    expect(tree).toMatchSnapshot();
});