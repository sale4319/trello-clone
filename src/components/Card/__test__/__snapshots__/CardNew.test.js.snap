// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mateches snapshot 1`] = `null`;
